app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -185,3 +263,131 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         }
     }
 }
+
+class HomeTileAdapter(
+        private val uiLifecycleCancelJob: Job,
+        private val tiles: MutableList<HomeTile>,
+        private val onUrlEnteredListener: OnUrlEnteredListener
+) : RecyclerView.Adapter<TileViewHolder>() {
+
+    override fun onBindViewHolder(holder: TileViewHolder, position: Int) = with (holder) {
+        val item = tiles[position]
+        when (item) {
+            is BundledHomeTile -> {
+                onBindBundledHomeTile(holder, item)
+                setLayoutMarginParams(iconView, R.dimen.bundled_home_tile_margin_value)
+            }
+            is CustomHomeTile -> {
+                onBindCustomHomeTile(uiLifecycleCancelJob, holder, item)
+                setLayoutMarginParams(iconView, R.dimen.custom_home_tile_margin_value)
+            }
+        }.forceExhaustive
+
+        itemView.setOnClickListener {
+            onUrlEnteredListener.onNonTextInputUrlEntered(item.url)
```

Comment:
Now that this is part of the overlay, you should use the overlay methods to "send" information to the browser, rather than doing it directly. So take a look at how the overlay does this.